packages/http-client-python/generator/pygen/codegen/serializers/__init__.py

```diff
@@ -159,7 +159,9 @@ def serialize(self) -> None:
                 self._serialize_and_write_top_level_folder(env=env, namespace=client_namespace)
 
             # add models folder if there are models in this namespace
-            if (client_namespace_type.models or client_namespace_type.enums) and self.code_model.options["models_mode"]:
+            if (
+                self.code_model.has_non_json_models(client_namespace_type.models) or client_namespace_type.enums
```

Comment:
When there are only json models which will not be serialized, we shall skip the serialization for models.

Reply:
can you explain this logic more clearly? I'm having a hard time following this and also only writing the models file below if there are non-json models

Reply:
Without the logic, when we try to generate SDK with flavor=unbranded for https://github.com/microsoft/typespec/blob/main/packages/http-specs/specs/parameters/basic/main.tsp, you will see empty _models like:
![image](https://github.com/user-attachments/assets/0fa62a07-5c84-4909-8296-183e5193b3f9)
The root cause is that its body type is json model which
![image](https://github.com/user-attachments/assets/73f164e8-00e9-45d3-aa41-629cc4f542ea)